itests/standalone/basic/src/main/java/org/wildfly/camel/test/cdi/CDIContextCreationIntegrationTest.java

```diff
@@ -61,56 +60,57 @@ public static JavaArchive createDeployment() {
-        final FakeContainer container = new FakeContainer();
-        Container.Instance.set(container);
+        FakeContextTracker tracker = new FakeContextTracker();
+        try {
+            tracker.open();
+
+            Assert.assertEquals(0, tracker.getContextCount());
+            Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
 
-        Assert.assertEquals(0, container.getContextCount());
-        Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+            deployer.deploy(CDI_CONTEXT_A);
+            Assert.assertEquals(1, tracker.getContextCount());
+            Assert.assertEquals(1, contextRegistry.getCamelContexts().size());
 
-        deployer.deploy(CDI_CONTEXT_A);
-        Assert.assertEquals(1, container.getContextCount());
-        Assert.assertEquals(1, contextRegistry.getCamelContexts().size());
+            deployer.deploy(CDI_CONTEXT_B);
+            Assert.assertEquals(2, tracker.getContextCount());
+            Assert.assertEquals(2, contextRegistry.getCamelContexts().size());
 
-        deployer.deploy(CDI_CONTEXT_B);
-        Assert.assertEquals(2, container.getContextCount());
-        Assert.assertEquals(2, contextRegistry.getCamelContexts().size());
+            deployer.undeploy(CDI_CONTEXT_A);
+            deployer.undeploy(CDI_CONTEXT_B);
 
-        deployer.undeploy(CDI_CONTEXT_A);
-        deployer.undeploy(CDI_CONTEXT_B);
+            Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+        } finally {
+            tracker.close();
+        }
+    }
 
-        Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
+    @Deployment(name = CDI_CONTEXT_A, managed = false, testable = false)
+    public static JavaArchive createCdiTestJar() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, CDI_CONTEXT_A);
+        archive.addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        archive.addPackage(InjectedContextBean.class.getPackage());
+        return archive;
+    }
+
+    @Deployment(name = CDI_CONTEXT_B, managed = false , testable = false)
+    public static JavaArchive createOtherCdiTestJar() {
+        JavaArchive archive = ShrinkWrap.create(JavaArchive.class, CDI_CONTEXT_B);
+        archive.addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml");
+        archive.addPackage(InjectedContextBean.class.getPackage());
+        return archive;
     }
 
-    private static final class FakeContainer implements Container {
+    private class FakeContextTracker extends CamelContextTracker {
         private List<CamelContext> camelContextList = new ArrayList<>();
 
         @Override
-        public void manage(CamelContext camelContext) {
+        public void contextCreated(CamelContext camelctx) {
             synchronized (camelContextList) {
-                camelContextList.add(camelContext);
+                if (!camelctx.getClass().getName().contains("Proxy")) {
```

Comment:
Do we really want to do this? We already have a [CamelContextTracker implementing this logic](https://github.com/jamesnetherton/wildfly-camel/blob/8ad839488088fd0b30f0c41192a971e7929cf88c/subsystem/core/src/main/java/org/wildfly/extension/camel/service/CamelContextRegistryService.java#L194-L196). Therefore, aren't these assertions one and the same thing?

```
Assert.assertEquals(0, tracker.getContextCount());
Assert.assertEquals(0, contextRegistry.getCamelContexts().size());
```